File: src/main/kotlin/com/wtw/demo/comparator/StringDiffer.java
package com.wtw.demo.comparator;

import java.util.Objects;
import java.util.Optional;

public class StringDiffer implements Differ<String> {

    private boolean ignoreCase = false;

    @Override
    public <S extends String> Optional<DiffResult> diff(S a, S b) {
        boolean comparison = Objects.equals(a,b);
        return comparison ?
                Optional.empty() :
                Optional.of(DiffResult.change(a, b));
    }
}

File: src/main/kotlin/com/wtw/demo/comparator/StringDiffer.kt
package com.wtw.demo.comparator

import java.util.*

class StringDiffer : Differ<String> {
    private val ignoreCase = false
    override fun diff(a: String?, b: String?): Optional<DiffResult> {
        val comparison = a == b
        return if (comparison) Optional.empty() else Optional.of(DiffResult.change(a, b))
    }
}
